pkg/blockstorage/awsefs/awsefs.go

```diff
@@ -285,16 +285,50 @@ func (e *efs) getBackupTags(ctx context.Context, arn string) (map[string]string,
 }
 
 func (e *efs) VolumeDelete(ctx context.Context, volume *blockstorage.Volume) error {
+	mts, err := e.getMountTargets(ctx, volume.ID)
+	if isVolumeNotFound(err) {
+		return nil
+	}
+	err = e.deleteMountTargets(ctx, mts)
+	if err != nil {
+		return errors.Wrap(err, "Failed to delete mount targets")
+	}
+
 	req := &awsefs.DeleteFileSystemInput{}
 	req.SetFileSystemId(volume.ID)
-
-	_, err := e.DeleteFileSystemWithContext(ctx, req)
+	_, err = e.DeleteFileSystemWithContext(ctx, req)
 	if isVolumeNotFound(err) {
 		return nil
 	}
 	return err
 }
 
+func (e *efs) getMountTargets(ctx context.Context, fsID string) ([]*awsefs.MountTargetDescription, error) {
+	mts := make([]*awsefs.MountTargetDescription, 0)
+	for resp, req := emptyResponseRequestForMountTargets(); resp.NextMarker != nil; req.Marker = resp.NextMarker {
+		var err error
+		req.SetFileSystemId(fsID)
+		resp, err = e.DescribeMountTargetsWithContext(ctx, req)
+		if err != nil {
+			return nil, err
+		}
+		mts = append(mts, resp.MountTargets...)
+	}
+	return mts, nil
+}
+
+func (e *efs) deleteMountTargets(ctx context.Context, mts []*awsefs.MountTargetDescription) error {
+	for _, mt := range mts {
+		req := &awsefs.DeleteMountTargetInput{}
+		req.SetMountTargetId(*mt.MountTargetId)
+		_, err := e.DeleteMountTargetWithContext(ctx, req)
```

Comment:
@hakanmemisoglu were can I find this function `DeleteMountTargetWithContext` definition?

Reply:
@SupriyaKasten 
It is defined in the anonymous field.

https://docs.aws.amazon.com/sdk-for-go/api/service/efs/#EFS.DeleteMountTargetWithContext